### ISSUE TITLE
Inconsistent behaviour when executing luarocks in different directories

### ISSUE THREAD

**Issue by wushu037** [luarocks/luarocks#1526]
**version: LuaRocks 3.9.2**

I didn't find a description of the problem I'm having in the [rocks usage documentation](https://github.com/luarocks/luarocks/wiki/Using-LuaRocks). So I don't know if luarocks is doing this on purpose or if it's due to a bug.

The situation is that executing luarocks commands in the luarocks installation directory (and any subdirectories) does not produce the same behaviour as executing luarocks commands in other directories.

For example:

- Executing luarocks install ... in the luarocks installation directory, the module will be installed in the luarocks installation directory /lua_modules. while executing this command in other paths, the module will be installed in /usr/local. The subpaths all conform to [rock's tree structure](https://github.com/luarocks/luarocks/wiki/Rocks-repositories).
- Only if you execute luarocks path in the luarocks installation directory, the resulting rules will contain entries that match the luarocks installation directory/lua_modules path.

It's as if the two directories are two different environments. If it's a bug, I hope it's fixed, and if it's intentional design, I'd like to know where the documentation related to it is?

**Comment by hishamhm** [comment]
Yes, the design is intentional. The documentation is unfortunately lacking (and help would be much appreciated!) but here is a summary from the original design document: https://github.com/luarocks/luarocks/wiki/Project:-LuaRocks-per-project-workflow
